imagebuildah/stage_executor.go

```diff
@@ -1131,6 +1131,8 @@ func (s *StageExecutor) commit(ctx context.Context, ib *imagebuilder.Builder, cr
 	}
 	s.builder.SetHostname(config.Hostname)
 	s.builder.SetDomainname(config.Domainname)
+	s.builder.SetArchitecture(s.executor.systemContext.ArchitectureChoice)
+	s.builder.SetOS(s.executor.systemContext.OSChoice)
```

Comment:
We shouldn't use `(*types.SystemContext).{Architecture,OS}Choice` for _creating_ an image. Those fields should only be used for _pulling_ an image (see https://github.com/containers/image/blob/master/types/types.go#L510).